pandas/tests/io/parser/test_dtypes.py

```diff
@@ -509,3 +509,27 @@ def test_numeric_dtype(all_parsers, dtype):
 
     result = parser.read_csv(StringIO(data), header=None, dtype=dtype)
     tm.assert_frame_equal(expected, result)
+
+
+def test_intna_precision(all_parsers):
+    parser = all_parsers
+    data = "1556559573141592653\n1556559573141592654\n\n1556559573141592655"
+    dtype = 'Int64'
+
+    expected = DataFrame([
+        [1556559573141592653],
+        [1556559573141592654],
+        [0],
+        [1556559573141592655],
+    ], dtype=dtype)
+    expected.iloc[2] = np.nan  # TODO: fix general bug on df construction
+
+    result = parser.read_csv(StringIO(data), header=None, dtype=dtype,
+                             skip_blank_lines=False)
+    
+    tm.assert_frame_equal(result, expected)
+
+    # See why tm.assert_frame_equal doesn't fail...
```

Comment:
Not sure I follow what's going here exactly?

Reply:
I didn't dig through the assert_*_methods but I'm pretty sure there is internal casting to float64 going on, so you can lose precision pretty easily. 

Here's an example that passes without issue:

```python
>>> ser1 = pd.Series([1556559573141592653, 1556559573141592654], dtype='Int64')
>>> ser2 = pd.Series([1556559573141592654, 1556559573141592654], dtype='Int64')
>>> tm.assert_series_equal(ser1, ser2) # should fail but doesn't
```

Reply:
Also going to need several more tests to check this behavior, but we can get revisit this point after the actual implementation has been critiqued.

Reply:
Totally. I think quite a few more need to go into test_integers for construction.

When I started this I wasn't expecting the precision issue with DataFrame construction and the assert methods. Ended up being a Pandora's box pretty quick, so posting this for discussion to see how we want to tackle the various components

Reply:
Hmm...that's awkward.  Should open an issue about this.  Our handling of large numbers is definitely bug-prone.

Reply:
FWIW I can get `assert_series_equal` to work if `check_exact=True` is passed.  My guess is that the float conversion happens due to defaulting to checking equality up to a given precision:
```python
In [1]: import pandas as pd; pd.__version__
Out[1]: '0.25.0.dev0+476.g9feb3ad92c'

In [2]: ser1 = pd.Series([1556559573141592653, 1556559573141592654], dtype='Int64')

In [3]: ser2 = pd.Series([1556559573141592654, 1556559573141592654], dtype='Int64')

In [4]: pd.util.testing.assert_series_equal(ser1, ser2)

In [5]: pd.util.testing.assert_series_equal(ser1, ser2, check_exact=True)
---------------------------------------------------------------------------
AssertionError: Series are different

Series values are different (50.0 %)
[left]:  [1556559573141592653, 1556559573141592654]
[right]: [1556559573141592654, 1556559573141592654]
```
I guess it could make sense to default to exact checks for integer dtypes instead of doing precision checks, but it doesn't really seem like a clean solution since it'd require implementing dtype specific behavior.